### PR TITLE
[TEAM-3] Unhappy path add experience

### PR DESCRIPTION
**Pull request by LixiaSun**
Unhappy path add experience

### CODE REVIEW

File: src/test/resources/features/apiaddexperience.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@ApiAddExperience
+Feature: add experience
+
+  @smoke @regression @api @unhappy
+  Scenario: add experience unhappy
+    Given [API] user is logged in
+      | username               | password |
+      | lixiasun2000@gmail.com | Lsun1688 |
+    Then user enters incorrect add experience inputs
```
Comment (Muradil-Erkin) [inline src/test/resources/features/apiaddexperience.feature:9]: - This step should just enter experience details, regardless if it's valid or invalid.
- All API steps should have `[API]` prefix

File: src/test/resources/features/apiaddexperience.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      | jobtitle | company | from |
+      | null     | null    | null |
+    Then user gets back error(s)
```
Comment (Muradil-Erkin) [inline src/test/resources/features/apiaddexperience.feature:12]: Expected errors should be 1 single field:
`| Title is required, Company is required, From date is required |`

File: src/test/resources/features/AddExperience.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    Given user is logged in
+      | username                         | password |
+      | muradil.erkin@boratechschool.com | Boratech |
+    When user clicks on the "Posts" navigation link
+    Then user should be navigated to the "Posts" page
+    When user enters the post comment
+      | comment                                          |
+      | Cucumber automation is a lot more than you think |
+    And user clicks on the submit button
+    Then user should see the success alert
```
Comment (Muradil-Erkin) [inline src/test/resources/features/AddExperience.feature:16]: This file should be deleted

File: src/test/java/runners/APIUnhappyPathRunner.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Suite
+@IncludeEngines("cucumber")
+@SelectClasspathResource("features")
+@ConfigurationParameter(key = PLUGIN_PROPERTY_NAME, value = "pretty, html:target/html-report/index.html")
+@ConfigurationParameter(key = GLUE_PROPERTY_NAME, value = "apiStepDefinitions")
+@ConfigurationParameter(key = ANSI_COLORS_DISABLED_PROPERTY_NAME, value = "true")
+@ConfigurationParameter(key = FILTER_TAGS_PROPERTY_NAME, value = "@unhappy")
+public class APIUnhappyPathRunner {
+
+}
```
Comment (Muradil-Erkin) [inline src/test/java/runners/APIUnhappyPathRunner.java:19]: This file can be deleted

File: src/test/java/apiStepDefinitions/Profile.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+public class Profile extends BoraAPI{
+	
+	 final static int expectedUnhappyStatusCode = 400;
+	
+	@Then("user enters incorrect add experience inputs")
+	public void user_enters_incorrect_add_experience_inputs(DataTable dataTable) {
+		Map<String, String> experience = dataTable.asMaps().get(0);
+		request.header("x-auth-token", login("lixiasun2000@gmail.com", "Lsun1688"));
+	   response = request.body(experience.keySet()).put("/api/profile/experience");
```
Comment (Muradil-Erkin) [inline src/test/java/apiStepDefinitions/Profile.java:17]: API implementation should be extracted to classes under `api/xxxx.java`

File: src/test/java/apiStepDefinitions/Profile.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+	@Then("user gets back error\\(s)")
+	public void user_gets_back_error_s(DataTable dataTable) {
+		Map<String, String> errors = dataTable.asMaps().get(0);
+		response = request.body(errors.keySet()).get("/api/profile/experience");
+		
+	   
+	}
+
+}
```
Comment (Muradil-Erkin) [inline src/test/java/apiStepDefinitions/Profile.java:29]: Files should be formatted

File: src/test/java/apiStepDefinitions/Profile.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.Map;
+
+import api.BoraAPI;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.Then;
+
+public class Profile extends BoraAPI{
+	
+	 final static int expectedUnhappyStatusCode = 400;
```
Comment (Muradil-Erkin) [inline src/test/java/apiStepDefinitions/Profile.java:11]: Constants should go under a file `utilities/constants.java`

**Review by Muradil-Erkin — CHANGES_REQUESTED** [review]

**Comment by LixiaSun** [comment]
Accidentally deleted the branch, and created  a new pull request.

**Pull request closed without merging**